packages/semantic-dom-diff/get-diffable-html.js

```diff
@@ -236,11 +236,35 @@ export function getDiffableHTML(html, options = {}) {
     }
   }
 
+  function removeComments(inputText) {
+    let outputText = '';
+    let index = 0;
+    let commentEnd;
+    let commentStart = inputText.indexOf('<!--', index);
+
+    while (commentStart !== -1) {
+      outputText += inputText.substring(index, commentStart);
+      commentEnd = inputText.indexOf('-->', commentStart);
+
+      if (commentEnd === -1) {
+        index = inputText.length();
+        commentStart = -1;
+      } else {
+        index = commentEnd + 3;
+        commentStart = inputText.indexOf('<!--', index);
+      }
+    }
+
+    outputText += inputText.substring(index);
+
+    return outputText;
+  }
+
   let container;
 
   if (typeof html === 'string') {
     container = document.createElement('diff-container');
-    container.innerHTML = html;
+    container.innerHTML = removeComments(html);
```

Comment:
doesn't this only handle the case of input as text? what about input as a node?  perhaps we can handle this while doing the node walk and we encounter a text node?

Reply:
You are right. I changed the implementation concatenating the sibling text nodes.